### PR TITLE
updated pipelines

### PR DESCRIPTION
**Pull request by behoyh**
As discussed, we will create a development and release build pipeline and a manual deployment pipeline for pre-production and production. This will only trigger on pull request creation and after a merge for development. For production, the trigger is set for only after an update to the master branch.

### CODE REVIEW

**Comment by behoyh** [comment]
> I see the azure-pipelines.yml creates a zip file for deployment then publishes it. That works. I'm assuming that this pipeline will be used for official builds then. We also need an additional pipeline for PR which doesn't generate and publish a zip file.
> 
> For the release pipeline, I don't see any changes. Shouldn't it deploy the zip file to the app service? I'm also expecting 2 release pipelines for each of the environments.

I added an additional pipeline for PRs. I need your help creating the key vault in Azure for the production build pipeline.

 The release pipelines are not housed in source control, but are available on the DevOps server. (The production release pipeline is not built yet) Additionally, by using a namespace, as this PR is doing, we can differentiate production releases from pre-production and deploy the latest of each environment.

**Comment by behoyh** [comment]
Renamed files for clarity.

File: azure-pipelines.prod.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+   - task: Npm@1
+     displayName: 'npm test'
+     inputs:
+       command: custom
+       workingDir: src
+       verbose: false
+       customCommand: 'run test'
+     env:
+       CUSTOMCONNSTR_DBCONNECTIONSTRING: $(hazzatdnn-ConnectionString)
```
Comment (fadyham) [inline azure-pipelines.prod.yml]: Prod pipeline should also publish the zip file just like develop

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,78 +1,46 @@
-# Node.js with React
-# Build a Node.js project that uses React.
-# Add steps that analyze code, save build artifacts, deploy, and more:
-# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
-
-trigger:
-- master
-- develop
+trigger: none
```
Comment (fadyham) [inline azure-pipelines.yml]: What does this pipeline do and when should it be used?

We're also missing the release pipeline which takes a zip file and publishes it to azure.

File: azure-pipelines.prod.yml (hunk 1 above)
Comment (behoyh) [reply azure-pipelines.prod.yml]: That's the part I need your help with, my current permissions don't allow me to create new resource targets.

Comment (behoyh) [reply azure-pipelines.prod.yml]: done

File: azure-pipelines.yml (hunk 2 above)
Comment (behoyh) [reply azure-pipelines.yml]: This pipeline run after changes have been pushed to master. The release pipleline is available [here](https://dev.azure.com/CopticDevs/hazzat-api/_release).

We still need to add a release pipeline, but that is not part of this PR.

File: azure-pipelines.develop.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    inputs:
+      SourceFolder: '$(Build.ArtifactStagingDirectory)'
+      Contents: '$(Build.BuildId).zip'
+      TargetFolder: '$(Build.ArtifactStagingDirectory)\ArtifactsToBePublished'
+
+  - task: PublishPipelineArtifact@1
+    displayName: 'Publish Artifact: drop'
+    inputs:
+      path: '$(Build.ArtifactStagingDirectory)\ArtifactsToBePublished'
+      artifact: HazzatBuildJS
```
Comment (fadyham) [inline azure-pipelines.develop.yml:67]: Add new line at end of file

Comment (fadyham) [inline azure-pipelines.develop.yml:67]: For prod it was called HazzatBuildProd
Perhaps we should call this HazzatBuildPPE

Comment (behoyh) [reply azure-pipelines.develop.yml]: done

Comment (behoyh) [reply azure-pipelines.develop.yml]: added

**Pull request merged**